src/main/java/com/scaler/lld/design/assignments/singleton/FileBasedConfigurationManagerImpl.java

```diff
@@ -2,49 +2,77 @@
 
 public class FileBasedConfigurationManagerImpl extends FileBasedConfigurationManager {
 
+    private static FileBasedConfigurationManager instance = null;
+
+    private  FileBasedConfigurationManagerImpl() {}
+
     @Override
     public String getConfiguration(String key) {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'getConfiguration'");
+        return getProperties().getProperty(key);
     }
 
     @Override
     public <T> T getConfiguration(String key, Class<T> type) {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'getConfiguration'");
+        String value = getProperties().getProperty(key);
+        if(value == null) {
+            return null;
+        }
+        if(type.isAssignableFrom(String.class)) {
+            return (T) value;
+        }
+        else if(type.isAssignableFrom(Integer.class)) {
+            return (T) Integer.valueOf(value);
+        }
+        else if(type.isAssignableFrom(Double.class)) {
+            return (T) Double.valueOf(value);
+        }
+        else if(type.isAssignableFrom(Float.class)) {
+            return (T) Float.valueOf(value);
+        }
+        else if(type.isAssignableFrom(Boolean.class)) {
+            return (T) Boolean.valueOf(value);
+        }
+        else {
+            return null;
+        }
```

Comment:
There is a `convert` method in the parent class which you can use instead of implementing the casting here.